docs/addons/argo-cd.md

```diff
@@ -122,25 +122,86 @@ The application promotion process in the above example is handled entirely throu
 
 The framework provides support to supply repository and administrator secrets in AWS Secrets Manager. This support is evolving and will be improved over time as ArgoCD itself matures. 
 
+### Private Repositories
+
 **SSH Key Authentication**
 
-1. Set `credentialsType` to `SSH` when defining `ApplicationRepository` in the ArgoCD add-on configuration.
-2. Define the secret in AWS Secret Manager as "Plain Text" that contains the private SSH key and (**important**) replicate it to all the desired regions. Please see [instructions for GitHub](https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh) on details on setting up SSH access.
+1. Set `credentialsType` to `SSH` when defining bootstrap repository in the ArgoCD add-on configuration.
+
+```typescript
+.addOns(new ssp.addons.ArgoCDAddOn({
+    bootstrapRepo: {
+        repoUrl: 'git@github.com:aws-samples/ssp-eks-workloads.git',
+        path: 'envs/dev',
+        credentialsSecretName: 'github-ssh-json',
+        credentialsType: 'SSH'
+    }
+}))
+```
+
+*Note:* In this case the configuration assumes that there is a secret `github-ssh-json` define in the target account and all the regions where the blueprint will be deployed. 
+
+2. Define the secret in AWS Secret Manager as "Plain Text" that contains a JSON structure (as of ArgoCD 2.x) with the fields `sshPrivateKey` and `url` defined. Note, that JSON does not allow line break characters, so all new line characters must be escaped with `\n`.
+
+Example Structure:
+```json
+{
+    "sshPrivateKey": "-----BEGIN THIS IS NOT A REAL PRIVATE KEY-----\nb3BlbnNzaC1rtdjEAAAAABG5vbmUAAAAEbm9uZQAAAAAAAAABAAACFwAAAAdzc2gtcn\nNhAAAAAwEAAQAAAgEAy82zTTDStK+s0dnaYzE7vLSAcwsiHM8gN\nhq2p5TfcjCcYUWetyu6e/xx5Rh+AwbVvDV5h9QyMw4NJobwuj5PBnhkc3QfwJAO5wOnl7R\nGbehIleWWZLs9qq`DufViQsa0fDwP6JCrqD14aIozg6sJ0Oqi7vQkV+jR0ht/\nuFO1ANXBn2ih0ZpXeHSbPDLeZQjlOBrbGytnCbdvLtfGEsV0WO2oIieWVXJj/zzpKuMmrr\nebPsfwr36nLprOQV6IhDDo\n-----END NOT A REAL PRIVATE KEY-----\n",
+
+    "url": "git@github"`
+}
+```
+Note explicit `\n` characters in the `sshPrivateKey`.
```

Comment:
I would put this in bold or include a notice sign so that people don't miss this